From 121fdf16a2ff4cf651c376a37eeca255544c47fd Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 27 Aug 2020 17:57:18 +0100
Subject: [PATCH] ARM: dts: bcm2835: Use the L2 non-allocating alias

The /soc/dma-ranges property on BCM2835 currently results in DMA
addresses in the range 0x40000000-0x5fffffff. This will allocate in the
system L2 cache, which may adversely affect performance.

Change the dma-ranges property to give addresses in the range
0x80000000-0x9fffffff, which are coherent with L2 but non-allocating.

See: https://github.com/raspberrypi/linux/issues/3602

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2835.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/arm/boot/dts/bcm2835.dtsi
+++ b/arch/arm/boot/dts/bcm2835.dtsi
@@ -18,7 +18,7 @@
 
 	soc {
 		ranges = <0x7e000000 0x20000000 0x02000000>;
-		dma-ranges = <0x40000000 0x00000000 0x20000000>;
+		dma-ranges = <0x80000000 0x00000000 0x20000000>;
 	};
 
 	arm-pmu {
